app/src/org/commcare/views/Combobox.java

```diff
@@ -0,0 +1,138 @@
+package org.commcare.views;
+
+import android.content.Context;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.TypedValue;
+import android.view.View;
+import android.widget.AutoCompleteTextView;
+
+import org.commcare.adapters.ComboboxAdapter;
+import org.commcare.adapters.PermissiveComboboxAdapter;
+import org.commcare.dalvik.R;
+import org.commcare.views.widgets.SpinnerWidget;
+
+import java.util.Vector;
+
+/**
+ * Custom view that builds upon Android's built-in AutoCompleteTextView and adds:
+ * -Validation and auto-correcting of user-entered text to only allow strings that are part of
+ * an available answer choice
+ * -Formatting to make the drop-down view look identical to that of CommCare's SpinnerWidget
+ * -The ability to add a custom adapter that implements custom filtering rules (rather than using
+ * the default filter of AutoCompleteTextView)
+ *
+ * @author Aliza Stone
+ */
+public class Combobox extends AutoCompleteTextView {
+
+    private Vector<String> choices;
+    private CharSequence lastAcceptableStringEntered = "";
+    private int lastValidCursorLocation;
+    private boolean fixingInvalidEntry;
+    private ComboboxAdapter customAdapter;
+
+    public Combobox(Context context, Vector<String> choices, boolean addEmptyFirstChoice,
+                    boolean permissive, int fontSize) {
+        super(context);
+        this.choices = choices;
+        setupAdapter(context, fontSize, addEmptyFirstChoice, permissive);
```

Comment:
It seems like passing in an adapter itself, and then setting the items in "Setup adapter" might be a more extensible way to manage this in the future if we need to allow for a few different matching styles?